ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -215,27 +215,35 @@ private[kvutils] class TransactionCommitter(
               value.getContractState.getContractKey -> Conversions.stateKeyToContractId(key)
           }
 
-        engine
-          .validate(
-            Set(transactionEntry.submitter),
-            SubmittedTransaction(transactionEntry.transaction),
-            transactionEntry.ledgerEffectiveTime,
-            commitContext.participantId,
-            transactionEntry.submissionTime,
-            transactionEntry.submissionSeed,
-          )
-          .consume(
-            lookupContract(transactionEntry, commitContext),
-            lookupPackage(transactionEntry, commitContext),
-            lookupKey(commitContext, knownKeys),
-          )
-          .fold(
-            err =>
-              reject[DamlTransactionEntrySummary](
-                commitContext.recordTime,
-                buildRejectionLogEntry(transactionEntry, rejectionReasonForValidationError(err))),
-            _ => StepContinue[DamlTransactionEntrySummary](transactionEntry)
-          )
+        try {
+          engine
+            .validate(
+              Set(transactionEntry.submitter),
+              SubmittedTransaction(transactionEntry.transaction),
+              transactionEntry.ledgerEffectiveTime,
+              commitContext.participantId,
+              transactionEntry.submissionTime,
+              transactionEntry.submissionSeed,
+            )
+            .consume(
+              lookupContract(transactionEntry, commitContext),
+              lookupPackage(transactionEntry, commitContext),
+              lookupKey(commitContext, knownKeys),
+            )
+            .fold(
+              err =>
+                reject[DamlTransactionEntrySummary](
+                  commitContext.recordTime,
+                  buildRejectionLogEntry(transactionEntry, rejectionReasonForValidationError(err))),
+              _ => StepContinue[DamlTransactionEntrySummary](transactionEntry)
+            )
+        } catch {
+          case err: Err.MissingInputState =>
+            logger.warn("Exception during model conformance validation.", err)
+            reject(
+              commitContext.recordTime,
+              buildRejectionLogEntry(transactionEntry, RejectionReason.Disputed(err.getMessage)))
```

Comment:
I think we should be more careful with the rejection reason. `Disputed` is meant to indicate a bug or malicious behaviour on the participant side. As you explained in the PR comment, we can also end up here if there's a contention issue, which should be reported as `Inconsistent`; see `rejectionReasonForValidationError` below.

Reply:
AFAICS only differences in contract key lookup should get us here with `Inconsistent`. So we could avoid the problem by checking in `validateContractKeys` that the contract keys of the input state resolve to the same contract IDs as those given in the input contracts.

Reply:
Agreed. Spoke to @gerolf-da and we'll do this separately, as this is blocking the DAML SDK 1.8.0 RC release.

Reply:
We will address this in two ways during contract key validation:

1. Check that the contract ID is in the declared inputs.
2. Check that contract is actually visible to the submitter.

If either of these fail, we'll reject with `Inconsistent`.

@andreaslochbihler-da, can you confirm this is what we agreed upon?

Reply:
Sounds good.